File: day6/main.js
const fs = require('fs');

const data = fs.readFileSync("in.txt", "utf8");
const mlen = 14; // Length of the marker

for (let i = 0;; i++) {

    const chars = data.slice(i, i + mlen);

    if (new Set(chars).size == mlen) {
        console.log(`First marker after ${i + mlen} characters`);
        break;
    }
}

File: day6/main.py
with open("in.txt", "r") as f:
    data = f.read()

    # Length of the marker with distinct characters 
    mlen = 14

    i = 0
    while True:
        chars = [*data[i:i+mlen]]
        if len(set(chars)) == mlen: break
        i += 1
    
    print(f"First marker after {i+mlen} characters")
